### PR TITLE
Doc changes for #877

### PR DESCRIPTION
**Pull request by coderanger**
Can someone double check that that's the right config syntax? Haven't played much with configurable plugins.

### CODE REVIEW

File: DOC_CHANGES.md [hunk 1]
```diff
@@ -7,3 +7,69 @@ Description of the required change.
...
+## Shard Seed Plugin
+
+The new `shard_seed` attribute provides a normally distributed integer ID value
+to all nodes. This value will be between 0 and 4294967295 (0xFFFFFFFF), based
+on a consistent hash of one or more pieces of node data. A consistent hash
+means that a node with the same source inputs will always result in the same
+shard seed, but collisions are possible.
+
+The sources used to compute the seed value can be configured either in your
+Chef config file or your Ohai config file ([ed: does anyone use Ohai config files?]):
```
Comment (jaymzh) [inline DOC_CHANGES.md]: you have to whenever you are developing Ohai. We need to make this better though.

File: DOC_CHANGES.md [hunk 2]
```diff
@@ -7,3 +7,69 @@ Description of the required change.
...
+
+```ruby
+Ohai::Config[:plugins][:shard][:sources] = [:fqdn]
```
Comment (jaymzh) [inline DOC_CHANGES.md]: s/shard/shard_seed/

File: DOC_CHANGES.md [hunk 3]
```diff
@@ -7,3 +7,69 @@ Description of the required change.
 -->
 
 # Ohai Doc Changes:
+
+## Shard Seed Plugin
+
+The new `shard_seed` attribute provides a normally distributed integer ID value
```
Comment (MarkGibbons) [inline DOC_CHANGES.md]: Hopefully it's not an approximately normal distribution.  I think md5 would be close to a uniform distribution though.
http://michiel.buddingh.eu/distribution-of-hash-values

File: DOC_CHANGES.md (hunk 1 above)
Comment (coderanger) [inline DOC_CHANGES.md]: I thought the whole point of splitting `chef-config` into its own gem was to allow stuff like Ohai to read Chef config files directly.

Comment (jaymzh) [inline DOC_CHANGES.md]: Oh that'd be neat. But now, run `ohai -c /etc/chef/client.rb` and watch it fail.

Comment (coderanger) [inline DOC_CHANGES.md]: Think it's worth mentioning ohai config files at all for normal users though? Probably a red herring for most people that are smart and don't touch ohai core :)

Comment (jaymzh) [inline DOC_CHANGES.md]: Believe or not I know several people who use Ohai outside of Chef, so yes.

File: DOC_CHANGES.md (hunk 2 above)
Comment (jaymzh) [inline DOC_CHANGES.md]: Also isn't those Ohai.config now?

Comment (jaymzh) [inline DOC_CHANGES.md]: (or Ohai.plugin for fun)

Comment (jaymzh) [inline DOC_CHANGES.md]: apparently it's "ohai.plugin" according to the docs. /me/sighs.

File: DOC_CHANGES.md [hunk 4]
```diff
@@ -7,3 +7,75 @@ Description of the required change.
...
+
+Two main use cases for the seed value are slow rollouts of new features and
+computing service IDs.
+
+To do slow rollouts, you can wrap new features or options in your recipe code
+with a conditional `if` statement like:
+
+```ruby
+# Enable feature for 20% of nodes.
+if (node['shard_seed'] % 100) < 20
```
Comment (jaymzh) [inline DOC_CHANGES.md]: fun side note: technically because we start at 0, you're this is 21% :)

**Comment by tas50** [comment]
@coderanger and @jaymzh Is this ready for merge?

**Comment by coderanger** [comment]
As long as the config file bit looks correct to people that know the Ohai config APIs, yes.

**Pull request merged**